sankey/sankey.js

```diff
@@ -107,16 +110,24 @@ d3.sankey = function() {
   // Nodes are assigned the maximum breadth of incoming neighbors plus one;
   // nodes with no incoming links are assigned breadth zero, while
   // nodes with no outgoing links are assigned the maximum breadth.
+  // nodes with xpos defined override the above rules
   function computeNodeBreadths() {
     var remainingNodes = nodes,
         nextNodes,
-        x = 0;
-
+        x = 0,
+		maxx = 0;
+	
     while (remainingNodes.length) {
       nextNodes = [];
       remainingNodes.forEach(function(node) {
-        node.x = x;
         node.dx = nodeWidth;
+		if(node.xpos != null) {
```

Comment:
has anyone tried using this if block? seems to go into an infinite loop somewhere...